src/common/meta/src/key.rs

```diff
@@ -215,9 +215,14 @@ impl TableMetadataManager {
     /// The caller MUST ensure it has the exclusive access to `TableNameKey`.
     pub async fn create_table_metadata(
         &self,
-        table_info: RawTableInfo,
+        mut table_info: RawTableInfo,
         region_routes: Vec<RegionRoute>,
     ) -> Result<()> {
+        let mut region_numbers = vec![];
```

Comment:
```suggestion
        let mut region_numbers = Vec::with_capacity(region_routes.len());
```

Or maybe we can use iterator to construct it:

```rust
let region_numbers = region_routes.iter()
    .map(|region| region.region.id.region_number())
    .collect::<Vec<_>>();
```

Reply:
@NiwakaDev Please apply this suggestion.

Reply:
Done: [f009a71](https://github.com/GreptimeTeam/greptimedb/pull/2209/commits/f009a711961ff79f64c692cf30ea6dd2c6ab10ca)

Please let me know if there are any omissions.